File: app/src/main/java/example/com/crackle/utils/MovieApiService.java
package example.com.crackle.utils;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

import static example.com.crackle.utils.Constants.BASE_URL;

public class MovieApiService {

    private static Retrofit retrofit;

    private static OkHttpClient.Builder httpClient = new OkHttpClient.Builder();
    private static HttpLoggingInterceptor loggingInterceptor = new HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY);

    public static Retrofit getClient () {

        if (retrofit == null) {

            httpClient.addInterceptor(loggingInterceptor);

            retrofit = new Retrofit.Builder()
                    .addConverterFactory(GsonConverterFactory.create())
                    .client(httpClient.build())
                    .baseUrl(BASE_URL)
                    .build();
        }

        return retrofit;
    }
}

File: app/src/main/java/example/com/crackle/utils/MovieApiService.kt
package example.com.crackle.utils

import example.com.crackle.listener.MovieApiClient
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object MovieApiService {

    private val loggingInterceptor = HttpLoggingInterceptor()
            .apply { level = HttpLoggingInterceptor.Level.BODY }
    private val httpClient = OkHttpClient.Builder()
            .addInterceptor(loggingInterceptor)
            .build()


    private val retrofit: Retrofit = Retrofit.Builder()
            .addConverterFactory(GsonConverterFactory.create())
            .client(httpClient)
            .baseUrl(Constants.BASE_URL)
            .build()

    val client: MovieApiClient = retrofit.create(MovieApiClient::class.java)
}
